commands/publish/index.js

```diff
@@ -84,6 +84,18 @@ class PublishCommand extends Command {
           `
         );
       }
+
+      if (!this.options.skipNpm) {
+        GitUtilities.fetchGitRemote(this.execOpts);
+        if (GitUtilities.isBehindUpstream(this.gitRemote, this.execOpts)) {
+          const remote = `${this.gitRemote}/${currentBranch} `;
+          this.logger.warn(dedent`
```

Comment:
This should be throwing a `new ValidationError("EBEHIND", msg)` with this message so it sets the `process.exitCode` to `1`, indicating an error status. A `return false` keeps the `process.exitCode` as the default `0`, which in unix-land means "no errors, everything is okay".

We never want lerna attempting to push a non-fast-forward commit/tag, after all. Instead of "Skipping publishing", maybe indicate what command to run to fix the problem? (Usually as simple as `git pull`, in my experience)

Reply:
Good point, I exited with `0` so that in a CI build scenario, the build would still be marked as successful if the publish step was aborted; but exiting with an error (and thus failing a build) is probably more accurate.